src/importer.js

```diff
@@ -0,0 +1,50 @@
+const fs = require('fs');
+const EventEmitter = require('events').EventEmitter;
+const csvjson = require('csvjson');
+
+const checkedFiles = [];
+
+class Importer extends EventEmitter {
+  constructor(dirWatcher, path) {
+    super();
+    // sync call:
+    // dirWatcher.on('data change', () =>
+    //   this.importSync(path)
+    // );
+    // ---------------
+    // async call:
+    dirWatcher.on('data change', () =>
+      this.import(path)
+        .then(data => console.log(data))
+        .catch(err => console.log(err)));
+  }
+
+  import(path) {
+    return new Promise((res, rej) => {
+      const files = fs.readdirSync(path);
+      const processedData = [];
+      files.forEach((file) => {
```

Comment:
Don't use forEach loops, because they based on side-effects. For synchronous code, or async/await functions with sequential order use for-of loops. For parallel work use `map` from array prototype together with `Promise.all()`